src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -432,6 +449,15 @@ ucs_status_t uct_tcp_sockcm_ep_recv(uct_tcp_sockcm_ep_t *cep)
     uct_tcp_sockcm_priv_data_hdr_t *hdr;
     ucs_status_t status;
 
+    /* if there is nothing left to read and got here due to an old/irrelevant read
+     * event, return */
+    if (ucs_test_all_flags(cep->state, UCT_TCP_SOCKCM_EP_SERVER_CREATED            |
+                                       UCT_TCP_SOCKCM_EP_SERVER_NOTIFY_CB_INVOKED) ||
+        ucs_test_all_flags(cep->state, UCT_TCP_SOCKCM_EP_ON_CLIENT                 |
+                                       UCT_TCP_SOCKCM_EP_DATA_RECEIVED) ) {
+        return UCS_OK;
+    }
```

Comment:
i don't think we need this check on recv side. we can see if uct_tcp_sockcm_ep_recv_nb actually received something or not. if it didn't (recv() returned EAGAIN), we can ignore